packages/core/src/App/app.jsx

```diff
@@ -47,10 +49,22 @@ const AppWithoutTranslation = ({ root_store }) => {
     const { preferred_language } = root_store.client;
     const { is_dark_mode_on } = root_store.ui;
     const is_dark_mode = is_dark_mode_on || JSON.parse(localStorage.getItem('ui_store'))?.is_dark_mode_on;
-    const is_dtrader_v2 =
-        isDTraderV2() && (location.pathname.startsWith(routes.trade) || location.pathname.startsWith('/contract/'));
     const language = preferred_language ?? getInitialLanguage();
 
+    const [dtrader_v2_enabled_gb] = useGrowthbookGetFeatureValue({
+        featureFlag: 'dtrader_v2_enabled',
+        defaultValue: false,
+    });
+    const [dtrader_v2_enabled, setDTraderV2Enabled] = React.useState();
+
+    React.useEffect(() => {
+        setDTraderV2Enabled(
+            (isDTraderV2() || (Boolean(dtrader_v2_enabled_gb) && isDTraderV2Width())) &&
```

Comment:
that "isDtraderV2Width" worries me a little bit - as really, nothing is retriggering explicitly that flag recalculation when width changes, 

so lets consider scenario:
- opened dtrater on low resolution, some hooks got executed and set V2 in rendered components
- someone resizes window, its still V2, as all the caomponents cached the v2 version
- someone clicks something, and new component pops up, window is > 600px so now one of the components are set to v2

Reply:
so basically - IMO the current setup is super prone to a really hard bugs,

Reply:
Now with hook approach the use of this function is almost removed. Now there is only one place where this function isDTraderV2Width is called. But it is called every time dtrader module is loaded, so no issue.

Reply:
and very time when some one resizes , for the switch to mobile width and vice versa our app-content.jsx get rerendered, So by luck or strategy our previous changes were working :-D